pyEpiabm/pyEpiabm/tests/test_sweep/test_spatial_sweep.py

```diff
@@ -45,77 +50,96 @@ def test__call__(self, mock_inf, mock_force, mock_poisson, mock_dist,
-        test_sweep.bind_population(self.pop)
+        test_sweep.bind_population(test_pop)
         test_sweep(time)
         self.assertTrue(self.cell_inf.person_queue.empty())
 
         # Add in another cell with a susceptible, but still
         # no infectors so no infection events.
-        self.pop.add_cells(1)
-        self.cell_susc = self.pop.cells[1]
-        self.cell_susc.add_microcells(1)
-        self.microcell_susc = self.cell_susc.microcells[0]
-        self.microcell_susc.add_people(1)
-        self.infectee = self.microcell_susc.persons[0]
-        mock_infectee.return_value = [self.infectee]
-
-        test_sweep.bind_population(self.pop)
+        test_pop.add_cells(1)
+        cell_susc = test_pop.cells[1]
+        cell_susc.add_microcells(1)
+        microcell_susc = cell_susc.microcells[0]
+        microcell_susc.add_people(1)
+        infectee = microcell_susc.persons[0]
+        mock_infectee.return_value = [infectee]
+        mock_choices.return_value = [cell_susc]
+
+        test_sweep.bind_population(test_pop)
         test_sweep(time)
-        self.assertTrue(self.cell_susc.person_queue.empty())
+        self.assertTrue(cell_susc.person_queue.empty())
 
         # Change infector's status to infected
         self.infector.update_status(InfectionStatus.InfectMild)
-        test_sweep.bind_population(self.pop)
+        test_sweep.bind_population(test_pop)
         test_sweep(time)
-        self.assertEqual(self.cell_susc.person_queue.qsize(), 1)
+        self.assertEqual(cell_susc.person_queue.qsize(), 1)
 
         # Check when we have an infector but no infectees
-        self.infectee.update_status(InfectionStatus.Recovered)
-        self.cell_susc.person_queue = Queue()
-        test_sweep.bind_population(self.pop)
+        infectee.update_status(InfectionStatus.Recovered)
+        cell_susc.person_queue = Queue()
+        test_sweep.bind_population(test_pop)
         test_sweep(time)
-        self.infectee.update_status(InfectionStatus.Susceptible)
+        infectee.update_status(InfectionStatus.Susceptible)
 
         # Check when all (one) nan in distance, won't call nan_to_num
         mock_dist.return_value = 0
 
-        self.cell_susc.person_queue = Queue()
-        test_sweep.bind_population(self.pop)
+        cell_susc.person_queue = Queue()
+        test_sweep.bind_population(test_pop)
         test_sweep(time)
         self.assertFalse(mock_nan.called)
-        self.assertEqual(self.cell_susc.person_queue.qsize(), 1)
+        self.assertEqual(cell_susc.person_queue.qsize(), 1)
 
         # All (one) valid distances also won't call nan_to_num
         mock_dist.return_value = 2
         Parameters.instance().do_CovidSim = True
-        self.cell_susc.person_queue = Queue()
-        test_sweep.bind_population(self.pop)
+        cell_susc.person_queue = Queue()
+        test_sweep.bind_population(test_pop)
         test_sweep(time)
         self.assertFalse(mock_nan.called)
```

Comment:
```suggestion
        mock_nan.assert_not_called()
```
I believe this is more stable?